server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -476,9 +477,11 @@ public boolean containsBrokenAnalysis(String field) {
      */
     public SearchLookup lookup() {
         if (this.lookup == null) {
+            SourceFieldMapper sourceMapper = mappingLookup == null ? null : (SourceFieldMapper) mappingLookup.getMapper("_source");
```

Comment:
I might just do `boolean isSynthetic =` here so you don't have to reason about "null means false"

Reply:
Or, rather, "null means true"